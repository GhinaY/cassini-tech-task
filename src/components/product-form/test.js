import mockListingsData from '../../utils/test-utils/mock-listings-data';
import ProductForm from './index';
import { render } from '@testing-library/react';

jest.mock('../../components/category-filter', () => ({
  __esModule: true,
  default: () => (<div>category filter</div>)
}));

const mockListingDetails = mockListingsData[0];
const mockOnSubmit = jest.fn();

function renderComponent(props) {
  return(
    render(
      <ProductForm {...props}/>
    )
  );
};

describe('ProductForm', () => {
  beforeEach(() => {
    jest.spyOn(global, 'fetch').mockResolvedValue({
      json: jest.fn().mockResolvedValue({ id: mockListingsData[0].id })
    })
  });

  it('renders correctly when existing listing data available', () => {
    const { asFragment } = renderComponent({ listingDetails: mockListingDetails, onSubmit: mockOnSubmit });
    
    expect(asFragment()).toMatchSnapshot();
  });
  
  it('renders correctly when no existing listing data available', () => {
    const { asFragment } = renderComponent({ onSubmit: mockOnSubmit });
    
    expect(asFragment()).toMatchSnapshot();
  });
});

// Definitely needs a lot more tests, including functional ones, 
// but in the interests of time I'll skip them for now.
