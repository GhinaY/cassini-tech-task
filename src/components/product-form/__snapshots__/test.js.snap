// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductForm renders correctly when existing listing data available 1`] = `
<DocumentFragment>
  <img
    alt="the product"
    class="ProductImage"
    src="https://fakestoreapi.com/img/81fPKd-2AYL._AC_SL1500_.jpg"
  />
  <form
    class="ProductDetails"
    id="listingDetailsForm"
  >
    <div>
      category filter
    </div>
    <label
      for="title"
    >
      Product title:
    </label>
    <input
      class="ProductTitle"
      name="title"
      type="text"
      value="Fjallraven - Foldsack No. 1 Backpack, Fits 15 Laptops"
    />
    <label
      for="price"
    >
      Product price:
    </label>
    <input
      name="price"
      type="number"
      value="109.95"
    />
    <label
      for="image"
    >
      Product price:
    </label>
    <input
      name="image"
      type="url"
      value="https://fakestoreapi.com/img/81fPKd-2AYL._AC_SL1500_.jpg"
    />
    <label
      for="description"
    >
      Product description:
    </label>
    <textarea
      name="description"
    >
      Your perfect pack for everyday use and walks in the forest. Stash your laptop (up to 15 inches) in the padded sleeve, your everyday
    </textarea>
    <button
      class="Button"
      type="submit"
    >
      Save
    </button>
  </form>
</DocumentFragment>
`;

exports[`ProductForm renders correctly when no existing listing data available 1`] = `
<DocumentFragment>
  <img
    alt="the product"
    class="ProductImage"
    src="https://fakeimg.pl/400x400?text=No+img"
  />
  <form
    class="ProductDetails"
    id="listingDetailsForm"
  >
    <div>
      category filter
    </div>
    <label
      for="title"
    >
      Product title:
    </label>
    <input
      class="ProductTitle"
      name="title"
      type="text"
      value=""
    />
    <label
      for="price"
    >
      Product price:
    </label>
    <input
      name="price"
      type="number"
      value=""
    />
    <label
      for="image"
    >
      Product price:
    </label>
    <input
      name="image"
      type="url"
      value="https://fakeimg.pl/400x400?text=No+img"
    />
    <label
      for="description"
    >
      Product description:
    </label>
    <textarea
      name="description"
    />
    <button
      class="Button"
      type="submit"
    >
      Save
    </button>
  </form>
</DocumentFragment>
`;
